File: Interview/2023·10·23途虎/1.ts
/**
 * 找出车辆可能的故障原因
 * @param keywords string字符串一维数组 故障原因关键词
 * @param faults string字符串二维数组 故障原因列表
 * @return string字符串一维数组
 */
export function findFaults(keywords: string[], faults: string[][]): string[] {
    const arr = faults.map((fault) => ({
        fault,
        cnt: fault.reduce(
            (cnt, v, i) => cnt + (i !== 0 && keywords.includes(v) ? 1 : 0),
            0
        ),
    }));

    arr.sort((a, b) => b.cnt - a.cnt);

    return arr.filter(v => v.cnt > 0).map(v => v.fault[0]);
}

File: Interview/2023·11·9虾皮/1.cpp
class Solution {
public:
    vector<int> moveZero(vector<int>& array) {
        int size = array.size();
        vector<int> arr;
        int cnt = 0;

        for (int k = 0; k < size; ++k) {
            if (array[k] != 0) {
                arr.push_back(array[k]);
            } else {
                ++cnt;
            }
        }

        for (int k = 0; k < cnt; ++k) {
            arr.push_back(0);
        }

        return arr;
    }
};
